skbio/stats/composition.py

```diff
@@ -2058,3 +2060,93 @@ def dirmult_ttest(
         "Reject null hypothesis",
     ]
     return res[col_order]
+
+
+def dirmult_lme(
+    formula, data, groups, reml=True, method=None, fit_kwargs=None, **kwargs
+):
+    r"""Fit a Dirichlet Multinomial linear mixed effects model.
+
+    The Dirichlet-multinomial distribution is a compound distribution that
+    combines a Dirichlet distribution over the probabilities of a multinomial
+    distribution. This distribution is used to model the distribution of
+    species abundances in a community.
+
+    To perform
+
+    This function uses ``MixedLM`` module from ``statsmodels.formula.api``
+
+    Parameters
+    ----------
+    formula : str or generic Formula object
+        The formula specifying the model
+    data : array_like
+        The data for the model. data must define __getitem__ with the keys in
+        the formula terms args and kwargs are passed on to the model
+        instantiation. E.g., a numpy structured or rec array, a dictionary,
+        or a pandas DataFrame.
+    groups : str
+        The column name in data that identifies the grouping variable
+    reml : bool
+        If true, fit according to the REML likelihood, else fit the standard
+        likelihood using ML.
+    method : str
+        Optimization method. Can be a scipy.optimize method name, or a list of such
+        names to be tried in sequence.
+        See https://docs.scipy.org/doc/scipy/tutorial/optimize.html for all options.
+    fit_kwargs : dict
+        Keyword arguments to pass to the model fit function.
+        See ``statsmodels.regression.mixed_linear_model.MixedLM.fit``
+    **kwargs
+        Additional keyword arguments to pass to the mixedlm function only.
+
+    Returns
+    -------
+    results : statsmodels.regression.mixed_linear_model.MixedLMResults
+        The fitted model results.
+        See ``statsmodels.regression.mixed_linear_model.MixedLMResults``
+
+    See Also
+    --------
+    statsmodels.formula.api.mixedlm
+    statsmodels.regression.mixed_linear_model.MixedLM
+
+    Notes
+    -----
+    pass
+
+    Examples
+    --------
+    >>> data = pd.DataFrame([[1, 0, 10.2], [1, 1, 12.5],
+    ...                     [1, 7, 14.8], [1, 14, 16.1]],
+    ...                     index=[0, 1, 2, 3],
+    ...                     columns=["patient_id", "time_point",
+    ...                             "response"])
+    >>> result = dirmult_lme("response ~ time_point", groups="patient_id", data=data)
```

Comment:
The function can take an R formula? That is cool!

Reply:
I think the formula here is also off. The `'response'` here should be each microbe. But because we are dealing with many (i.e. hundreds / thousands) of microbes, it means that we are going to need to fix hundreds / thousands of MixedLM models. 

The `dirmult_lme` shouldn't make the user specify `response`, since it should be implicit with the input table.

Reply:
Hello Jamie, thank you for your feedback. I was previously unaware of this aspect, but now I understand that `response` would be each microbe. For the latter half, could you please give me any tips or recommendations on how to implement this change?